src/libs/nav2_robot/include/nav2_robot/ros_robot.hpp

```diff
@@ -37,16 +39,29 @@ class RosRobot : public Robot
 
   void enterSafeState() override;
 
-  geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr getCurrentPose();
+  bool getCurrentPose(geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr & robot_pose);
```

Comment:
@mhpanah @crdelsey 
What's the rationale to prefer a success/failure return code in this case versus an exception? We should be consistent with our error handling mechanisms, if possible, and I thought we were preferring exceptions. That is, we'd introduce specific exception types instead of using the generic ones, such as runtime_error.